### ISSUE TITLE
WGS mode

### ISSUE THREAD

**Issue by mxiong2** [nch-igm/snvstory#9]
Can snvstory take WGS data? Would it be enable by adjusting the defualt --mode WES to WGS?

**Comment by mxiong2** [comment]
Also, snvstory does not seem to write anything to the --output-dir "/data/my_folder" directory. For VCF files with multiple samples, no UMAP plots are generated for the samples. Is this normal?

Thanks.

**Comment by andreirajkovic** [comment]
> Can snvstory take WGS data? Would it be enable by adjusting the defualt --mode WES to WGS?

Yes just set it WGS and it will use a slightly different model to perform the analysis

> Also, snvstory does not seem to write anything to the --output-dir "/data/my_folder" directory. For VCF files with multiple samples, no UMAP plots are generated for the samples. Is this normal?
> 
> Thanks.


Is your docker command mounting a volume like so?

docker run -v /mnt/c/Users/andrei.rajkovic/PycharmProjects/snvstory/dev/data:/data -t "igm/churchill-ancestry:dev1" --path /data/HG00096_extracted.vcf.gz --resource /data/resource_dir --output-dir /data/output --genome-ver 38 --mode WES

**Comment by andreirajkovic** [comment]
@mxiong2 
An example of what the output directory structure looks like
![image](https://github.com/nch-igm/snvstory/assets/34891295/ba867c22-4ae8-4167-8c23-9ab2eddebd05)

**Comment by mxiong2** [comment]
I make a folder in the snvstory/snvstory/dev/data/my_output_data directory. snvstory does not seem to write anything to the output directory but it does write the results folder in the snvstory/snvstory/dev/data directory. Do I have to mount a directory when I run docker-compose run ancestry? 

For vcf files with multiple samples, does snvstory produce umap plots for the individual samples or do I need to run snvstory for each sample? 

I run this command: 

docker-compose run ancestry \

    --path "/data/vcf_files/wgs/test.vcf" \

    --resource "/data/resource_dir" \

    --output-dir "/data/my_output_data \

    --output_filename "test2" \

    --genome-ver 38 \

    --mode WGS

**Comment by andreirajkovic** [comment]
I think I see the issue. If you want it to write out to a specific location you'll need to mount that directory or modify the docker compose file which assumes the location of your root is in ./dev/data.

```bash
docker run \
-v /path/to/your/data:/data \
-v /path/to/your/output:/output \
-t "igm/churchill-ancestry:latest" \
--path /data/HG00096_extracted.vcf.gz \
--resource /data/resource_dir \
--output-dir /output \
--genome-ver 38 \
--mode WES
```

Here is an idea for how to do it in the docker-compose instead:
![image](https://github.com/nch-igm/snvstory/assets/34891295/35dc7bc8-739b-4a58-a654-57a6f1d3fd70)

**Comment by mxiong2** [comment]
I modify my code to ( with output folder call **out_dir**):

docker run \

-v /media/sf_MolecularNGS/Mai/snvstory/snvstory/dev/data:/data \

-t "igm/churchill-ancestry:latest" \

--path /data/vcf_files/wgs/test.vcf \

--resource /data/resource_dir \

--output-dir /data/out_dir \

--genome-ver 38 \

--mode WGS


I get this traceback error that the file already exists

![Capture](https://github.com/nch-igm/snvstory/assets/44978940/74c5f948-e367-497a-bc4e-0fb60df0d8c7)

**Comment by andreirajkovic** [comment]
Well, there is a small bug here. The flex_output function is misbehaving on line 118 the basename operation returns nothing, which leads to the bug above. I'll put a fix in for this now

**Comment by andreirajkovic** [comment]
Here is an example of the command I ran to make this work

```bash
docker run \
  -v /path/to/snvstory/dev/data:/data \
  -v /path/to/snvstory/dev/output:/output \
  -t "heritagehelix/snvstory:3.0.2" \
  --path /data/HG00096_extracted.vcf.gz \
  --resource /data/resource_dir \
  --output-dir /output \
  --genome-ver 38 \
  --mode WES
```
